pkg/plugin/keyvault.go

```diff
@@ -102,29 +123,45 @@ func newKeyVaultClient(
 		keyVersion:       keyVersion,
 		vaultURL:         *vaultURL,
 		azureEnvironment: env,
+		keyID:            fmt.Sprintf("%x", sha256.Sum256([]byte(fmt.Sprintf("%s/keys/%s/%s", *vaultURL, keyName, keyVersion)))),
+		algorithm:        kv.RSA15,
 	}
 	return client, nil
 }
 
-func (kvc *keyVaultClient) Encrypt(ctx context.Context, cipher []byte) ([]byte, error) {
+func (kvc *keyVaultClient) Encrypt(ctx context.Context, cipher []byte) (*service.EncryptResponse, error) {
 	value := base64.RawURLEncoding.EncodeToString(cipher)
 
 	params := kv.KeyOperationsParameters{
-		Algorithm: kv.RSA15,
+		Algorithm: kvc.algorithm,
 		Value:     &value,
 	}
 	result, err := kvc.baseClient.Encrypt(ctx, kvc.vaultURL, kvc.keyName, kvc.keyVersion, params)
 	if err != nil {
 		return nil, fmt.Errorf("failed to encrypt, error: %+v", err)
 	}
-	return []byte(*result.Result), nil
+
+	annotations := map[string][]byte{
+		dateAnnotationKey:           []byte(result.Header.Get(dateAnnotationValue)),
+		requestIDAnnotationKey:      []byte(result.Header.Get(requestIDAnnotationValue)),
+		keyvaultRegionAnnotationKey: []byte(result.Header.Get(keyvaultRegionAnnotationValue)),
+		versionAnnotationKey:        []byte(encryptionResponseVersion),
+		algorithmAnnotationKey:      []byte(kvc.algorithm),
```

Comment:
Shouldn't this come from AKV and not be hard coded?

Reply:
We need to tell KV which algo to use while Encrypting - https://github.com/Azure/kubernetes-kms/blob/e2b14860744f254872b170e4f2b6ac3de6f32d8d/pkg/plugin/keyvault.go#L112-L115

Reply:
Does AKV return the algorithm used as part of the encryption response?  That would make more sense to use.

Reply:
I opened https://github.com/Azure/kubernetes-kms/issues/234 to track this.